lib/firestore/models.dart

```diff
@@ -53,20 +56,153 @@ abstract class Reference {
 }
 
 class CollectionReference extends Reference {
-  CollectionReference(FirestoreGateway gateway, String path)
-      : super(gateway, path) {
+  final StructuredQuery _structuredQuery = StructuredQuery();
+  final FirestoreGateway gateway;
+
+  /// Constructs a [CollectionReference] using [FirestoreGateway] and path.
+  ///
+  /// Throws [Exception] if path contains odd amount of '/'.
+  CollectionReference(this.gateway, String path) : super(gateway, path) {
+    _structuredQuery.from
+        .add(StructuredQuery_CollectionSelector()..collectionId = id);
     if (_fullPath.split('/').length % 2 == 1) {
       throw Exception('Path is not a collection: $path');
     }
   }
 
+  CollectionReference whereIsEqualTo(String fieldPath, dynamic value) {
+    _createFilter(fieldPath, null);
```

Comment:
Shouldn't this be `value` and not `null`?

Reply:
Oopsies